### DISCUSSION TITLE
How to join the same table from different tables?

### DISCUSSION THREAD

**Discussion by ihernandezvelarde** [atoti/atoti#752] [Q&A]
Hi!

I would like to know how to have the following structure of 3 tables connected to each other without causing the conflict of not knowing where to access the information.

![image](https://user-images.githubusercontent.com/91269912/219082992-bdf810ba-c8fa-430f-95f8-cd257c743450.png)

## Steps to reproduce
We have Table1 with:

- id,field1,field2,another
          
Table2 with:

- field,value,another

And Table3 with:

- field,another1,extra

![image](https://user-images.githubusercontent.com/91269912/219079975-a7688be1-4626-4424-821f-e46666756502.png)

The tables contain these inputs:

TABLE 1
id,field1,field2,another          
id1,value11,value12,another1
id2,value21,value22,another2
id3,value31,value32,another3

TABLE 2
field,value,another
value11,1,another1
value21,2,another2

TABLE 3
field,another1,extra
value23,another1,wanted1
value33,another2,wanted2

## Actual Result
![image](https://user-images.githubusercontent.com/91269912/219083093-d8ad74d7-e287-481a-afde-20e77bed57bc.png)

## Expected Result
![image](https://user-images.githubusercontent.com/91269912/219082992-bdf810ba-c8fa-430f-95f8-cd257c743450.png)

## Environment

- atoti: 0.7.2
- Python: 3.9.13
- Operating System: Windows 10 Enterprise

**Comment by tibdex** [comment]
Hi,

"diamond" joins are not supported at the moment. The alternative is to duplicate Table3:

```py
table3 = session.read_csv("table3.csv", table_name="TABLE 3", keys=["FIELD"])
table3_prime = session.read_csv("table3.csv", table_name="TABLE 3 PRIME", keys=["FIELD"])

table1.join(table3, ...)
table2.join(table3_prime, ...)
```

**Reply by ihernandezvelarde** [reply]
Hi,
Thanks for your response, there are any plans to put it in future versions the "diamond" joins ?

**Reply by tibdex** [reply]
It's not planned for now.

It's hard to tell what you're trying to do with the generic table and column names you gave. Maybe there is another solution (e.g. accessing the column of TABLE3 from TABLE1 through TABLE2). Can you explain a bit more the use case?
